cypher/cypher-docs/src/docs/dev/query-tuning.asciidoc

```diff
@@ -57,6 +90,40 @@ When Cypher is building execution plans, it looks at the schema to see if it can
 These index decisions are only valid until the schema changes, so adding or removing indexes leads to the execution plan cache being flushed.
 
 
+[[cypher-preparser-runtime]]
+=== Cypher runtime
+
+Using the execution plan, the query is executed -- and records returned -- by the query engine, or _runtime_.
+Depending on whether Neo4j Enterprise Edition or Neo4j Community Edition is used, there are three different runtimes available:
+
+Interpreted::
+In this runtime, the operators in the execution plan are chained together in a tree, where each non-leaf operator feeds from one or two child operators.
+The tree thus comprises nested iterators, and the records are streamed in a pipelined manner from the top iterator, which pulls from the next iterator and so on.
+
+Slotted::
+This is very similar to the interpreted runtime, except that there are additional optimizations regarding the way in which the records are streamed through the iterators.
+This results in improvements to both the performance and memory usage of the query.
+In effect, this can be thought of as a the 'faster interpreted' runtime.
+
+[NOTE]
+The slotted runtime is only available in Neo4j Enterprise Edition.
+
+Compiled::
+Algorithms are employed to group intelligently the operators in the execution plan in order to generate new combinations and orders of execution which are optimised for performance and memory usage.
+While this should lead to superior performance in most cases (over both the interpreted and slotted runtimes), it is still under development.
+
+[NOTE]
+The compiled runtime is only available in Neo4j Enterprise Edition.
+
+[options="header"]
+|===
+| Preparser option | Description | Default?
+| `runtime=interpreted` | This will force the query to use the interpreted runtime.  | This is the default option for some queries in Enterprise Edition, and the only option for all queries in Community Edition (so it is not necessary to use this option in Community Edition).
+| `runtime=slotted` | This will force the query to use the slotted runtime. | This is the default option for some queries in Enterprise Edition only. Setting this option for a query that does not support the slotted runtime will cause the query to fall back to using the interpreted runtime.
+| `runtime=compiled` | This will force the query to use the compiled runtime. | This is the default option for some queries in Enterprise Edition only. Setting this option for a query that does not support the compiled runtime will cause the query to fall back to using the slotted or interpreted runtime.
```

Comment:
As in my 3.2 comments, I think the word 'force' is too strong for the fallback process. I would prefer that we have a separate paragraph describing the fallback process, and then in this table only describe how these options simply set at which level in that process to start.

Reply:
👍